### PR TITLE
模型

### PR DESCRIPTION
**Pull request by conghaowoooong**
- [x] Overview
- [x] Style Hypothesis
  - [x] Agent Representation
  - [x] Behavior Features
  - [x] Style Hypothesis
- [x] Stylized Prediction
  - [x] Agent Representation
  - [x] Interaction Features
  - [x] Multi-Style Prediction
- [x] Loss Functions

### CODE REVIEW

File: satoshi_aaai/contents/method.tex [hunk 1]
```diff
@@ -157,90 +142,76 @@ \subsubsection{Agent Representation}
...
 
-We provide two kinds of \BETAMODEL, the deterministic \BETAMODEL-D and the generative \BETAMODEL-G, to handle different prediction scenarios:
+We provide two kinds of \MODEL, \MODELD and \MODELG to handle different prediction scenarios.
 
-\paragraph{a. Deterministic \MODELD}
+\paragraph{a. \MODELD}
 
-We employ an MLP (called MLP$_D$) To obtain the deterministic 2D-coordinate prediction sequences.
-We apply it on the last dimension of the multi-style interaction features $h_{\beta}$ to obtain the deterministic prediction $\hat{Y}_D$.
+We employ an MLP (D$_D$ in \TABLE{tab_method}) to obtain the deterministic 2D-coordinate prediction sequences.
```
Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:164]: D$_D$ - D$_d$ 公式里是 d

File: satoshi_aaai/contents/method.tex [hunk 2]
```diff
@@ -157,90 +142,76 @@ \subsubsection{Agent Representation}
...
     \end{aligned}
 \end{equation}
-
-Finally, it outputs the interaction features $h_{\beta} = \{{h_\beta}_k\}_k \in \mathbb{R}^{K_c \times t_f \times d}$ that contains rich interaction representations and all $K_c$ stylized conditions.
+Finally, it outputs the multi-style interaction features $h_{\beta} = \{{h_\beta}_k\}_k \in \mathbb{R}^{K_c \times t_f \times d}$.
 
 \subsubsection{Multi-Style Prediction}
 
-We provide two kinds of \BETAMODEL, the deterministic \BETAMODEL-D and the generative \BETAMODEL-G, to handle different prediction scenarios:
+We provide two kinds of \MODEL, \MODELD and \MODELG to handle different prediction scenarios.
```
Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:160]: We provide two kinds of \MODEL, \MODELD and \MODELG, to handle different prediction scenarios.
\MODELG后面需加一个逗号，这样,...,才是\MODEL的同位语

File: satoshi_aaai/contents/method.tex [hunk 3]
```diff
@@ -157,90 +142,76 @@ \subsubsection{Agent Representation}
...
 
-\paragraph{a. Deterministic \MODELD}
+\paragraph{a. \MODELD}
 
-We employ an MLP (called MLP$_D$) To obtain the deterministic 2D-coordinate prediction sequences.
-We apply it on the last dimension of the multi-style interaction features $h_{\beta}$ to obtain the deterministic prediction $\hat{Y}_D$.
+We employ an MLP (D$_D$ in \TABLE{tab_method}) to obtain the deterministic 2D-coordinate prediction sequences.
+We apply it on the last dimension of the multi-style interaction features $h_{\beta}$ to obtain the multi-style deterministic prediction $\hat{Y}_D$.
```
Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:165]: on - to

File: satoshi_aaai/contents/method.tex [hunk 4]
```diff
@@ -157,90 +142,76 @@ \subsubsection{Agent Representation}
...
 \end{equation}
 So far, all the sub-networks will learn how to classify and generate $K_c$ predictions with different behavior styles, and then give agents all styles of deterministic predictions $\hat{Y}_D$.
-That is the multi-style deterministic \MODELD.
 
-\paragraph{b. Generative \MODELG}
+\paragraph{b. \MODELG}
 
-The deterministic models actually give the ``average'' prediction for each input \cite{socialGAN}.
+The \MODELD~gives the ``average''\cite{socialGAN} prediction for each hidden behavior category.
 It lacks strong generation capabilities to show agents' uncertainty or randomness when making decisions.
```
Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:175]: strong - powerful

File: satoshi_aaai/contents/method.tex [hunk 5]
```diff
@@ -157,90 +142,76 @@ \subsubsection{Agent Representation}
...
 \begin{equation}
-    \mathcal{L} = \mu_1 \mathcal{L}_{\mbox{ad}} + \mu_2 \mathcal{L}_{\mbox{fd}} + \mu_3 \mathcal{L}_{\mbox{diff}} + \mu_4 \mathcal{L}_{kl}.
+    \mathcal{L} = \mu_1 \mathcal{L}_{\mbox{ad}} + \mu_2 \mathcal{L}_{\mbox{c}} + \mu_3 \mathcal{L}_{\mbox{diff}} + \mu_4 \mathcal{L}_{kl}.
 \end{equation}
-In the above equation, $\mathcal{L}_{\mbox{ad}}$, $\mathcal{L}_{\mbox{fd}}$, and $\mathcal{L}_{\mbox{diff}}$ denotes the \emph{Average Displacement Loss}, the \emph{Final Displacement Loss}, and the \emph{Differential Displacement Loss}.
+In the above equation, $\mathcal{L}_{\mbox{ad}}$, $\mathcal{L}_{\mbox{c}}$, and $\mathcal{L}_{\mbox{diff}}$ denote the \emph{Average Displacement Loss}, the \emph{Categorical Loss}, and the \emph{Differential Displacement Loss}.
 $\{\mu_1, \mu_2, \mu_3\}$ are a set of balance coefficients.
-Besides, the $\mathcal{L}_{kl}$ indicates the KL Divergence item, and it is only applied in \MODELG~($\mu_4 = 1$).
-For deterministic \MODELD, we set $\mu_4 = 0$.
+Besides, the $\mathcal{L}_{kl}$ indicates the KL Divergence item, and it is only applied in \MODELG.
```
Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:195]: 这下标出现了kl，与149的会不会让人confuse

File: satoshi_aaai/contents/method.tex [hunk 6]
```diff
@@ -157,90 +142,76 @@ \subsubsection{Agent Representation}
...
 
 \subsubsection{Average Displacement Loss}
-$\mathcal{L}_{\mbox{ad}}$ is the average point-wise displacement for each 2D prediction point with their groundtruths.
-We use this item to let the \BETAMODEL~learn the overall trend of the forecast trajectory.
+$\mathcal{L}_{\mbox{ad}}$ is the average point-wise displacement for each predicted point with their groundtruths.
```
Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:198]: their groundtruths - the ground truth
前面用的point

Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:198]: their groundtruths - the ground truth
前面用的point

Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:198]: their groundtruths - the ground truth
前面用的point

File: satoshi_aaai/contents/method.tex [hunk 7]
```diff
@@ -157,90 +142,76 @@ \subsubsection{Agent Representation}
...
     \mathcal{L}_{\mbox{ad}} = \frac{1}{t_f} \sum_t \Vert y^i_t - {\hat{y}}^i_t \Vert_2.
 \end{equation}
 
-\subsubsection{Final Displacement Loss}
-$\mathcal{L}_{\mbox{fd}}$ is the average minimum point-wise displacement among each 2D destination hypothesises and their true destination.
-It is used to train variables in \ALPHAMODEL.
-The minimum operation is used to train each hidden behavior category's prediction network.
-See details in \EQUA{eq_alpha_loss1} and \EQUA{eq_alpha_loss2}.
+\subsubsection{Categorical Loss}
+$\mathcal{L}_{\mbox{c}}$ is used to train each hidden behavior category's prediction network, making them generate multi-style hypothesises.
```
Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:205]: hypotheses

File: satoshi_aaai/contents/method.tex [hunk 8]
```diff
@@ -157,90 +142,76 @@ \subsubsection{Agent Representation}
...
 \begin{equation}
-    \mathcal{L}_{\mbox{fd}} = \frac{1}{N} \sum_{i=1}^N \min_k \Vert D^i_k - d^i \Vert_2 = \frac{1}{N} \sum_{k=1}^{K_c} N_k \mathcal{L}_d (k),
+    \mathcal{L}_{\mbox{c}} = \frac{1}{N} \sum_{k=1}^{K_c} N_k \mathcal{L}_d (k).
 \end{equation} 
-where $D^i_k$ denotes the $k$-th hidden category's destination hypothesis given by \ALPHAMODEL~for agent-$i$, and $d^i$ represents his groundtruth last position during the prediction period.
+See details in \EQUA{eq_alpha_loss1} and \EQUA{eq_alpha_loss2}.
 
-\subsubsection{Differential Displacement Loss}\footnote{
+\subsubsection{Differential Displacement Loss\footnote{
     Due to the space limitation, please see the experimental analysis of this loss function in Appendix.
```
Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:213]: in Appendix - in the appendix
与前面保持一致

File: satoshi_aaai/contents/method.tex [hunk 9]
```diff
@@ -257,11 +228,11 @@ \subsubsection{Differential Displacement Loss}\footnote{
 
 \subsubsection{KL Loss}
 $\mathcal{L}_{kl}$ is the KL divergence between the distribution of agent features $h_\beta$ (denote by $P_\beta$) and the normalized Gaussian distribution $N(0, I)$.
```
Comment (NorthOcean) [inline satoshi_aaai/contents/method.tex:230]: denote - denoted

File: satoshi_aaai/contents/method.tex (hunk 5 above)
Comment (conghaowoooong) [reply satoshi_aaai/contents/method.tex:195]: 线性预测换成了上标

File: satoshi_aaai/contents/method.tex (hunk 8 above)
Comment (conghaowoooong) [reply satoshi_aaai/contents/method.tex:213]: 换成了`in Appendix`
